### PR TITLE
Add export user data flow

### PR DESCRIPTION
**Pull request by tonisevener**
**Phabricator:** 
https://phabricator.wikimedia.org/T258212

### Notes
This feature allows users to export their app container via a `UIActivityViewController` from the Help and Feedback screen.

### Test Steps
1. Fresh install, log in, enable syncing via app Settings > Article storage and syncing.
2. Create some reading lists and save some articles.
3. Go to App Settings > Help and feedback.
4. Tap "Export User Data" button at the bottom
5. After a while, you should see a share activity sheet pop up. Airdrop the zip file to your machine.
6. Open the zip file, confirm you see data (console.log, Permanent Cache directory, User Data Export Sync Info json file, and some Wikipedia.sqlite database files).

### CODE REVIEW

File: Wikipedia/Code/HelpViewController.swift [hunk 1]
```diff
@@ -38,13 +56,43 @@ class HelpViewController: SinglePageWebViewController {
...
     
+    enum UserDataExportError: Error {
+        case unableToDetectSystemFreeSpace
+        case unableToDetectLogSize
+    }
+    
+    @objc func exportUserData() {
+        
+        let confirmationTitle = WMFLocalizedString("export-user-data-confirmation-title", value: "Share app library?", comment: "Title of confirmation modal after user taps \"Export User Data\" button.")
+        let confirmationMessage = WMFLocalizedString("export-user-data-confirmation-message", value: "Sharing your app library includes data on your Reading lists and history, preferences and Explore feed content. This data file should only be shared with a trusted recipient to use for technical diagnostic purposes.", comment: "Message of confirmation modal after user taps \"Export User Data\" button.")
```
Comment (staykids) [inline Wikipedia/Code/HelpViewController.swift:88]: I saw it in the Phab ticket, but I'd recommend a minor copy update to this string:

`Sharing your app library includes data about your Reading lists and history, preferences, and Explore feed content. This data file should only be shared with a trusted recipient to use for technical diagnostic purposes.`

File: Wikipedia/Code/HelpViewController.swift [hunk 2]
```diff
@@ -60,3 +108,240 @@ class HelpViewController: SinglePageWebViewController {
...
+        let apiController = readingListsController.apiController
+        let appSettingsShowSavedReadingList = readingListsController.isDefaultListEnabled
+        let appSettingsSyncSavedArticlesAndLists = readingListsController.isSyncEnabled
+        
+        let dispatchQueue = DispatchQueue.global(qos: .userInitiated)
+        dispatchQueue.async {
+
+            let sharedCache = SharedContainerCache<UserDataExportSyncInfo>.init(pathComponent: .userDataExportSyncInfo, defaultCache: { UserDataExportSyncInfo(serverReadingLists: [], serverReadingListEntries: [], appSettingsSyncSavedArticlesAndLists: false, appSettingsShowSavedReadingList: false) })
+            
+            apiController.getAllReadingLists { (serverReadingLists, since, error) in
```
Comment (staykids) [inline Wikipedia/Code/HelpViewController.swift:164]: Can change `since` and `error` to `_` because they are not used.

**Review by staykids — APPROVED** [review]
Nice - only a couple of minor notes if you also feel they are worth it, otherwise happy to just merge.

**Comment by tonisevener** [comment]
@staykids Agreed - this is done!

**Review by staykids — APPROVED** [review]

**Pull request merged**